src/lib/y2packager/product_finder.rb

```diff
@@ -0,0 +1,139 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2019 SUSE LLC, All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+# ------------------------------------------------------------------------------
+
+require "solv"
+
+module Y2Packager
+  # This class finds products in a Solv pool
+  class ProductFinder
+    #
+    # Constructor
+    #
+    # @param pool [Solv::Pool] the pool used for evaluating the products
+    #
+    def initialize(pool)
+      @pool = pool.pool
+    end
+
+    #
+    # Evaluate all products in the pool and return details about them
+    # including the dependencies.
+    #
+    # @param base [String] The name of the base product used for evaluating the
+    #  dependencies.
+    #
+    # @return [Array<Hash>] The list of found products
+    #
+    def products(base = nil)
+      marked_base_products = base_product_tags
+      # evaluate all products
+      pool.whatprovides(pool.str2id("product()")).each_with_object([]) do |s, list|
+        # the dependant repositories, includes also the transient dependencies
+        required = []
+        solver = pool.Solver
+        # select this product solvable (the product *-release package)
+        jobs = [pool.Job(Solv::Job::SOLVER_SOLVABLE | Solv::Job::SOLVER_INSTALL, s.id)]
+
+        if base
+          # select the base product
+          base_product = pool.select("product(#{base})", Solv::Selection::SELECTION_PROVIDES)
+          jobs += base_product.jobs(Solv::Job::SOLVER_INSTALL)
+        end
+
+        # run the solver to evaluate all dependencies
+        problems = solver.solve(jobs)
+
+        # in case of problems, ignore the dependencies
```

Comment:
why ignore in case of problems?

Reply:
Because we do not exactly know what was the problem. Maybe a missing dependency or conflict or ..??? And it would be strange to ask the user to resolve a conflict *before* adding the repository actually.